### DISCUSSION TITLE
What happens with multiple groups selected when hardlinking?

### DISCUSSION THREAD

**Discussion by dsrtusr88** [qarmin/czkawka#1149] [Q&A] [answered]
I recently discovered I have my hardlinking misconfigured resulting in tons of dupes.  In czkawka, if I have multiple groups selected and then hit hardlink, what would happen?  each group collapses into their own single hardlink or every file selected will hardlink (and ruin my library?)

**Comment by qarmin** [comment]
Each group is processed in isolation.

In each group, first selected item is treated as "original" and rest as duplicates(but this rarely matters in harlinking)

**Comment by dsrtusr88** [comment]
@qarmin 

so in the case:

Group 1
- item A
- item B

Group 2
- item C
- item D

If I have all of those selected at the same time and click `hardlink`, the result will be item A & item B hardlinked with the original dataset of item A persisting with the dataset of item B is removed.... and the same happens with item C and item D with two distinct datasets (A and C) remaining hardlinked to their respective files?

**Reply by qarmin** [reply, marked as answer]
B will be replaced with hardlink to A
D will be replaced with hardlink to C

Nothing else should happen
